dbms/src/Flash/Statistics/TableScanImpl.cpp

```diff
@@ -43,19 +43,32 @@ void TableScanStatistics::collectExtraRuntimeDetail()
     {
         for (const auto & io_stream : it->second)
         {
-            if (auto * cop_stream = dynamic_cast<CoprocessorBlockInputStream *>(io_stream.get()); cop_stream)
+            auto * cop_stream = dynamic_cast<CoprocessorBlockInputStream *>(io_stream.get());
+            // In tiflash_compute node, TableScan will be converted to ExchangeReceiver.
+            auto * exchange_stream = dynamic_cast<ExchangeReceiverInputStream *>(io_stream.get());
+            if (cop_stream || exchange_stream)
             {
-                for (const auto & connection_profile_info : cop_stream->getConnectionProfileInfos())
+                const std::vector<ConnectionProfileInfo> * connection_profile_infos = nullptr;
+                if (cop_stream)
+                    connection_profile_infos = &cop_stream->getConnectionProfileInfos();
+                else if (exchange_stream)
+                    connection_profile_infos = &exchange_stream->getConnectionProfileInfos();
+
+                for (const auto & connection_profile_info : *connection_profile_infos)
                 {
-                    cop_table_scan_detail.packets += connection_profile_info.packets;
-                    cop_table_scan_detail.bytes += connection_profile_info.bytes;
+                    remote_table_scan_detail.packets += connection_profile_info.packets;
+                    remote_table_scan_detail.bytes += connection_profile_info.bytes;
                 }
             }
             else if (auto * local_stream = dynamic_cast<IProfilingBlockInputStream *>(io_stream.get()); local_stream)
             {
                 /// local read input stream also is IProfilingBlockInputStream
                 local_table_scan_detail.bytes += local_stream->getProfileInfo().bytes;
             }
+            else
```

Comment:
`DAGStorageInterpreter` may produce `NullBlockInputStream` which is not `IProfilingBlockInputStream`

Reply:
D